core/src/main/java/org/apache/accumulo/core/client/ClientSideIteratorScanner.java

```diff
@@ -134,15 +142,22 @@ public SamplerConfiguration getSamplerConfiguration() {
       return samplerConfig;
     }
 
-    @Deprecated(since = "2.1.0")
     @Override
-    public ServiceEnvironment getServiceEnv() {
-      return new ClientServiceEnvironmentImpl(context.get());
+    @Deprecated(since = "2.0.0")
+    public AccumuloConfiguration getConfig() {
+      var ctx = context.get();
+      try {
+        return new ConfigurationCopy(
+            ctx.tableOperations().getConfiguration(ctx.getTableName(tableId.get())));
+      } catch (AccumuloException | TableNotFoundException e) {
+        throw new RuntimeException("Error getting table configuration", e);
+      }
```

Comment:
What is the rationale for implementing `getConfig`, and specifically as a copy of the table configuration? The behavior of this method in the interface was never well-defined, and was using an internal API and should never be called by user code. I'm not sure why it needs an implementation here, rather than just throwing an exception, like the deprecated default method does in the interface.

If we really do need this, the default method in the interface should implement it by calling the methods on `getPluginEnv()` and wrapping it similarly to what you did here, instead of that default method throwing an exception.

Reply:
I implemented it here since I thought that because it was deprecated but not yet removed, there should still be functionality for the method. The other impls of `IteratorEnvironment` still have this implemented; assumed same should be the case here. Less importantly, this also made testing the different impls of `IteratorEnvironment` in `IteratorEnvIT` more straight forward as each would call `getConfig()`, `getServiceEnv()`, and `getPluginEnv()`. If this shouldn't be implemented here, I can remove it, or I could change the default impl to use `getPluginEnv()` like you said. Whichever you think would be better.